packages/contracts/contracts/plugin/PluginRepo.sol

```diff
@@ -126,93 +144,93 @@ contract PluginRepo is IPluginRepo, Initializable, UUPSUpgradeable, ACL, Adaptiv
             });
         }
 
-        uint256 versionIdx = nextVersionIndex;
+        uint256 versionIndex = nextVersionIndex;
         nextVersionIndex = _uncheckedIncrement(nextVersionIndex);
-        versions[versionIdx] = Version(_newSemanticVersion, _pluginFactoryAddress, _contentURI);
-        versionIdxForSemantic[semanticVersionHash(_newSemanticVersion)] = versionIdx;
-        versionIdxForContract[_pluginFactoryAddress] = versionIdx;
+        versions[versionIndex] = Version(_newSemanticVersion, _pluginFactory, _contentURI);
+        versionIndexForSemantic[semanticVersionHash(_newSemanticVersion)] = versionIndex;
+        versionIndexForPluginFactory[_pluginFactory] = versionIndex;
 
-        emit NewVersion(versionIdx, _newSemanticVersion);
+        emit VersionCreated(versionIndex, _newSemanticVersion);
     }
 
-    /// @notice get latest plugin
-    /// @return semanticVersion Semantic version for latest pluginRepo version
-    /// @return pluginFactoryAddress Address of latest plugin factory for version
-    /// @return contentURI External URI for fetching latest version's content
-    function getLatest()
+    /// @notice Gets the version information of the latest version.
+    /// @return semanticVersion The semantic version number.
+    /// @return pluginFactory The address of the plugin factory associated with the version.
+    /// @return contentURI The external URI pointing to the content of the version.
+    function getLatestVersion()
         public
         view
         returns (
             uint16[3] memory semanticVersion,
-            address pluginFactoryAddress,
+            address pluginFactory,
             bytes memory contentURI
         )
     {
-        return getByVersionId(nextVersionIndex - 1);
+        return getVersionById(nextVersionIndex - 1);
     }
 
-    /// @notice get latest by plugin factory address
-    /// @return semanticVersion Semantic version for pluginRepo version
-    /// @return pluginFactoryAddress Address of plugin factory for version
-    /// @return contentURI External URI for fetching version's content
-    function getLatestForContractAddress(address _pluginFactoryAddress)
+    /// @notice Gets the version information associated with a plugin factory address.
+    /// @return semanticVersion The semantic version number.
+    /// @return pluginFactory The address of the plugin factory associated with the version.
+    /// @return contentURI The external URI pointing to the content of the version.
+    function getVersionByPluginFactory(address _pluginFactory)
```

Comment:
This has a different meaning. Latest != Version. I would revert this. :warning:

Reply:
@Rekard0 Could you check here plz?

Reply:
This function seems to always return `latest`.  So it should be `getLatestVersionByPluginFactory`.

Reply:
The naming of this function in aOS was wrong - it does not return the latest version, but the version associated with the `pluginFactory`. I discussed this with @Rekard0 on a call.

Reply:
I explained this in our chat: https://discord.com/channels/@me/989112020655820840/1001892671037657228 so we can close this conversation.